docs/DeveloperGuide.md

```diff
@@ -1,38 +1,575 @@
+
+These operations include:
+- Adding or removing Days to the Programme
+- Adding or removing Exercises to Days in the Programme
+- Updating the details of Exercises in Days in the Programme
+
+Given below is an example usage scenario for 'delete exercise' and how the edit programme functions at each step.
+
+Step 1. The user creates a programme with a given number of Days with their respective Exercises. ProgrammeList will contain a reference to this programme after its creation.
+
+![](images/editCommandStepOne.png)
+
+Step 2. The user executes `programme edit /p 1 /d 1 /x 1` to delete the first exercise in the first day of the first programme. The programme first retrieves the given day with `ProgrammeList#getDay()`.
+
+![](images/editCommandStepTwo.png)
+
+Step 3. With the Day object, it performs the `Day#deleteExercise()` with the given exercise ID
+
+![](images/editCommandStepThree.png)
+
+### Add Meal
+
+#### Feature Implementation
+The **Add Meal** feature manages the functionality related to adding meals to a daily record. It interacts with various components such as `History`, `DailyRecord`, and `MealList` to ensure meals are added correctly.
+
+The **Add Meal** command navigates through the following hierarchy:
+- **History** → **DailyRecord** → **MealList**
+- If a `DailyRecord` does not exist for a given date, it is created before adding the meal.
+- Similarly, a new `Meal` object is created and added to the `MealList` if it doesn't already exist.
+
+These operations include:
+- Adding meals to a `DailyRecord` in the `History`.
+- Updating existing meals in the `MealList` within a `DailyRecord`.
+- Displaying the result of the command execution.
+
+Given below is an example usage scenario for adding a meal and how the add meal command functions at each step.
+
+#### Example Usage Scenario
+
+**Step 1**: The user starts by adding a meal using the command:
+
+meal add /n [mealName] /c [calories]
+
+- The command is parsed and translated into an `AddMealCommand` object, which contains the necessary details.
+
+![Add Meal Step 1](images/AddMeal_Step_one_diagram.png)
+
+**Step 2**: The command retrieves the `DailyRecord` for the specified date from the `History` using `History#getRecordByDate()`. If no record exists, a new one is created.
+
+![Add Meal Step 2](images/AddMeal_Step_two_diagram.png)
+
+**Step 3**: The `AddMealCommand` adds the meal to the `MealList` of the `DailyRecord`. If the meallist already exists, it updates the existing meallist instead.
+
+![Add Meal Step 3](images/AddMeal_Step_three_diagram.png)
+
+**Step 4**: The newly added `Meal` object is returned to the `AddMealCommand` to display as part of the `CommandResult`.
+
+![Add Meal Step 4](images/AddMeal_Step_four_diagram.png)
+
+The overall design that enables this functionality is described generically by the following sequence diagram.
+
+#### Sequence Diagram for "Add Meal" Command
+
+![Add Meal Sequence Diagram](images/AddMeal_Sequence_diagram.png)
```

Comment:
![image](https://github.com/user-attachments/assets/12eb7689-dfb8-4dfd-b4a5-1f04f88678d8)
The add meal sequence diagram is missing an activation bar in MealList